web/tests/survey-form-spec.js

```diff
@@ -0,0 +1,109 @@
+describe('surveysForm directive', function() {
+    const getResponse = {any: 'Any'};
+    const postData = {other: 'Other'};
+    const fields = {thing: 'Thig'}; 
+
+    let scope;
+    let elm;
+    let formResponse;
+
+    let formUrl = 'http://from-url';
+    let responseUrl = 'http://response-url';
+
+    let $rootScope;
+    let $compile;
+    let $q;
+    let compileDirective;
+    let API;
+    let FieldsParser;
+
+    beforeEach(function () {
+        fixture.setBase('tests/api-description');
+
+        angular.module('formly', []);
+        angular.mock.module({
+            formlyConfig: {
+                setType: angular.noop
+            }
+        });
+        angular.mock.module('incuna-surveys.form-directive');
+
+        inject(function(_$rootScope_, _$compile_, _$q_, _API_, _FieldsParser_) {
+            $rootScope = _$rootScope_;
+            $compile = _$compile_;
+            $q = _$q_;
+            API = _API_;
+            FieldsParser = _FieldsParser_;
+            scope = $rootScope.$new();
+            scope.formUrl = formUrl;
+            scope.responseUrl = responseUrl;
```

Comment:
We normally use `this.variable` because the testing frameworks always clears the `this` object before each test, so it minimises the possibility of anything creeping through the tests